src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -0,0 +1,357 @@
+import { marketData } from 'state/slices/marketDataSlice/marketDataSlice'
+
+import { FOX_TOKEN_CONTRACT_ADDRESS, WETH_TOKEN_CONTRACT_ADDRESS } from './constants'
+import {
+  farmOpportunities,
+  FoxEthLpEarnOpportunityType,
+  FoxFarmingEarnOpportunityType,
+  lpOpportunity,
+} from './foxEthCommon'
+import { fetchPairData } from './utils'
+
+type FoxEthState = {
+  farmOpportunities: FoxFarmingEarnOpportunityType[]
+  lpOpportunity: FoxEthLpEarnOpportunityType
+}
+
+const initialState: FoxEthState = {
+  farmOpportunities,
+  lpOpportunity,
+}
+
+const moduleLogger = logger.child({ namespace: ['foxEthSlice'] })
+
+export const foxEth = createSlice({
+  name: 'foxEth',
+  initialState,
+  reducers: {
+    clear: () => initialState,
+    upsertLpOpportunity: (state, action: PayloadAction<Partial<FoxEthLpEarnOpportunityType>>) => {
+      state.lpOpportunity = {
+        ...state.lpOpportunity,
+        ...action.payload,
+      }
+    },
+    upsertFarmingOpportunity: (
+      state,
+      action: PayloadAction<Partial<FoxFarmingEarnOpportunityType>>,
+    ) => {
+      const stateOpportunityIndex = state.farmOpportunities.findIndex(
+        opportunity => opportunity.contractAddress === action.payload.contractAddress,
+      )
+      state.farmOpportunities[stateOpportunityIndex] = {
+        ...state.farmOpportunities[stateOpportunityIndex],
+        ...action.payload,
+      }
+    },
+  },
+})
+
+type GetFoxEthLpGeneralDataReturn = {
+  tvl: string
+  apy: string
+}
+type GetFoxEthLpWalletDataReturn = {
+  underlyingFoxAmount: string
+  underlyingEthAmount: string
+  cryptoAmount: string
+  fiatAmount: string
+}
+
+type GetFoxFarmingContractGeneralDataReturn = {
+  expired: boolean
+} & GetFoxEthLpGeneralDataReturn
+
+type GetFoxFarmingContractGeneralDataArgs = {
+  contractAddress: string
+}
+
+type GetFoxFarmingContractWalletDataReturn = {
+  cryptoAmount: string
+  fiatAmount: string
+  unclaimedRewards: string
+}
+
+type GetFoxFarmingContractWalletDataArgs = {
+  contractAddress: string
+  ethWalletAddress: string
+}
+
+type GetFoxEthLpWalletDataArgs = {
+  ethWalletAddress: string
+}
+
+export const foxEthApi = createApi({
+  reducerPath: 'foxEthApi',
+  ...BASE_RTK_CREATE_API_CONFIG,
+  endpoints: build => ({
+    getFoxEthLpGeneralData: build.query<GetFoxEthLpGeneralDataReturn, void>({
+      queryFn: async (_args, injectedStore) => {
+        try {
+          const { getState, dispatch } = injectedStore
+          const state: any = getState() // ReduxState causes circular dependency
+          const ethPrecision = state.assets.byId[ethAssetId].precision
+          const lpAssetPrecision = state.assets.byId[foxEthLpAssetId].precision
+          const ethPrice = state.marketData.crypto.byId[ethAssetId].price
+          const ethersProvider = getEthersProvider()
+          const uniV2LPContract = new Contract(
+            UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+            IUniswapV2Pair.abi,
+            ethersProvider,
+          )
```

Comment:
This will run every query and make Infura calls when contract instantiation really only needs to run once, so we will want for this to live at module-level instead